tests/analysis/java/test_book_server.py

```diff
@@ -0,0 +1,33 @@
+import os
+import subprocess
+
+from utils import constants
+from utils.command import build_analysis_command
+from utils.common import verify_triggered_rules, extract_rules
+from utils.report import assert_story_points_from_report_file, get_json_from_report_output_file
+
+
+def test_book_server_analysis(book_server_data):
+    application_data = book_server_data["source_analysis_on_bookserverapp"]
+    application_path = os.path.join(
+        os.getenv(constants.PROJECT_PATH),
+        'data',
+        'applications',
+        application_data['file_name']
+    )
+
+    command = build_analysis_command(
+        application_data['file_name'],
```

Comment:
From where are you getting application URL like 
https://github.com/ibraginsky/book-server ?

Reply:
Also `book_server.json` contains field `filename` which leads to particular directory where `book-server` app is kept